clang/lib/Driver/Driver.cpp

```diff
@@ -5202,6 +5252,18 @@ class OffloadingActionBuilder final {
       addInputs(sycl_device_wrapper_libs);
       if (isSpirvAOT || TC->getTriple().isNVPTX())
         addInputs(sycl_device_fallback_libs);
+
+      bool nativeBfloatLibs;
+      bool needBfloatLibs = selectBfloatLibs(TC, nativeBfloatLibs);
+      if (needBfloatLibs &&
+          TC->getTriple().getSubArch() != llvm::Triple::SPIRSubArch_fpga) {
```

Comment:
The fpga check is unnecessary given your explicit `spir64_gen` check in `selectBfloatLibs`

Reply:
Yes, that was an oversight.